routers/web/repo/issue.go

```diff
@@ -3106,6 +3106,11 @@ func UpdateCommentContent(ctx *context.Context) {
 		return
 	}
 
+	if comment.Issue.RepoID != ctx.Repo.Repository.ID {
```

Comment:
Just use `GetCommentForIssueByID(...)` ?

Reply:
It will check repository id but not issue id. So I leave it for future PRs but fixed other two.